spanner/value.go

```diff
@@ -49,13 +50,55 @@ const (
 	NumericScaleDigits = 9
 )
 
+// NumericLossHandlingMode describes the mode of how to deal with loss of
+// precision on numeric values.
+type NumericLossHandlingMode int
+
+const (
+	// NumericRound rounds an invalid numeric value, e.g., 0.1234567895 rounds
+	// to 0.123456790.
+	NumericRound NumericLossHandlingMode = iota
+	// NumericError throws an error when meeting invalid numeric value.
+	NumericError
+)
+
+// NumericPrecisionLossHandling is the configuration for hanlding loss of
+// precission on numeric values.
+var NumericPrecisionLossHandling NumericLossHandlingMode
+
 // NumericString returns a string representing a *big.Rat in a format compatible
 // with Spanner SQL. It returns a floating-point literal with 9 digits after the
 // decimal point.
 func NumericString(r *big.Rat) string {
 	return r.FloatString(NumericScaleDigits)
 }
 
+// validateNumeric returns nil if there are no errors. It will return an error
+// when the numeric number is not valid.
+func validateNumeric(r *big.Rat) error {
+	if r == nil {
+		return nil
+	}
+	// Add one more digit to the scale component to find out if there are more
+	// digits than required.
+	strRep := r.FloatString(NumericScaleDigits + 1)
+	strRep = strings.TrimRight(strRep, "0")
+	strRep = strings.TrimLeft(strRep, "-")
+	s := strings.Split(strRep, ".")
+	if len(s) != 2 {
+		return fmt.Errorf("invalid numeric float string: %s", strRep)
+	}
+	whole := s[0]
+	scale := s[1]
+	if len(scale) > NumericScaleDigits {
+		return fmt.Errorf("max scale for a numeric is %d. The requested numeric has more", NumericScaleDigits)
```

Comment:
nit:
```suggestion
		return fmt.Errorf("max scale for a numeric is %d. The requested numeric has %d", NumericScaleDigits, len(scale))
```

Reply:
This is not quite right. If the number has more than 10 digits after the radix point, `scale` will also be 10 as well because I limit it to 10 when I convert it to a float string: 

```go
strRep := r.FloatString(NumericScaleDigits + 1)
```

Reply:
Ah, good point.